status/condition_set.go

```diff
@@ -91,7 +91,7 @@ func (c ConditionSet) Get(t string) *Condition {
 
 // Set sets or updates the Condition on Conditions for Condition.Type.
 // If there is an update, Conditions are stored back sorted.
-func (c ConditionSet) Set(condition Condition) {
+func (c ConditionSet) Set(condition Condition) (modified bool) {
```

Comment:
feels weird to name the return value if you're not going to return the named variable. Wouldn't the better way to inform what the function is supposed to return with a comment?

Reply:
I prefer self documenting code than prose documentation

Reply:
We also need to revisit the comments on this whole package at some point. Not sure how go docgen works with these.